src/data/p5.TypedDict.js

```diff
@@ -0,0 +1,371 @@
+    return this.data[key];
+  }else{
+    console.log(key + ' does not exist in this Dictionary');
+  }
+};
+
+/**
+ * Changes the value of key if in Dictionary otherwise makes new one
+ *
+ *
+ */
+p5.TypedDict.prototype.set = function(key, value) {
+  if (arguments.length === 2) {
+    if(!this.data.hasOwnProperty(key)){
+      this.create(key, value);
+    } else {
+      this.data[key] = value;
+    }
+  }
+};
+
+
+p5.TypedDict.prototype._addObj = function(obj) {
+  for (var key in obj) {
+    if(this._validate(obj[key])) {
+      this.data[key] = obj[key];
+      this.count++;
+    }
+  }
+};
+
+
+/**
+ * Creates a new key-value pair in Dictionary object
+ *
+ *
+ */
+p5.TypedDict.prototype.create = function() {
+  if(arguments.length === 1) {
+    if(arguments[0] instanceof Object) {
+      this._addObj(arguments[0]);
+    }
+  }
+  else if(arguments.length === 2) {
+    var key = arguments[0];
+    var value = arguments[1];
+    if (this._validate(value)) {
+      this.data[key] = value;
+      this.count++;
+    } else {
+      console.log('those values dont work for this dictionary type');
+    }
+  } else {
+    console.log('In order to create a new Dictionary entry you must pass ' +
+      'an object or a key, value pair');
+  }
+};
+
+/**
+ * Empties out Dictionary object
+ *
+ *
+ */
+p5.TypedDict.prototype.clear = function(){
+  for(var key in this.data) {
+    delete this.data[key];
+  }
+  this.count = 0;
+};
+
+/**
+ * Removes a key-value pair in the Dictionary
+ *
+ *
+ */
+p5.TypedDict.prototype.remove = function(key) {
+  if(this.data.hasOwnProperty(key)) {
+    delete this.data[key];
+    this.count--;
+  } else {
+    throw key + ' does not exist in this Dictionary';
+  }
+};
+
+
+/**
+ * Logs the list of items currently in the Dictionary
+ *
+ *
+ */
+p5.TypedDict.prototype.print = function() {
+  for (var item in this.data) {
+    console.log('key:' + item + ' value:' + this.data[item]);
+  }
+};
+
+p5.TypedDict.prototype._validate = function(key, value) {
+  return true;
+};
+
+// /**
+//  *
+//  * A simple Dictionary class for Strings.
+//  *
+//  *
+//  * @class p5.StringDict
+//  * @constructor
+//  * @extends p5.Element
+//  * @param {String}
+//  */
+p5.StringDict = function(key, value) {
+  this.data = {};
+  this.count = 0;
+
+  if(arguments.length === 1) {
+    if(key instanceof Object) {
+      this.data = key;
+      this.count = Object.keys(key).length;
+    }
+  } else {
+    this.data[key] = value;
+    this.count++;
+  }
+  return this;
+};
+
+
+p5.StringDict.prototype = Object.create(p5.TypedDict.prototype);
+
+
+p5.StringDict.prototype._validate = function(value) {
+  return (typeof value === 'string');
+};
+
+p5.StringDict.prototype.constructor = p5.StringDict;
+
+// /**
+//  *
+//  * A simple Dictionary class for Numbers.
+//  *
+//  *
+//  * @class p5.NumberDict
+//  * @constructor
+//  * @extends p5.TypedDict
+//  * @param
+//  */
+p5.NumberDict = function(key, value) {
```

Comment:
In the latest commit, I used apply to set call the `TypedDict` constructor which works but creates a certain degree of messiness with the passing of arguments. I tried other approaches like `Object.create` and setting their constructor to the TypedDict constructor but they weren't retaining the variables made by the constructor like `data`. Not sure if there is a more elegant way.